### PR TITLE
buildscript: iterate example folder and build

### PR DESCRIPTION
**Pull request by YifeiZhuang**
cc. @temawi

### CODE REVIEW

**Comment by YifeiZhuang** [comment]
It seems revealing failure on `example-servlet` that was not covered by the current build.

```
+ pushd examples/example-servlet
+ ../gradlew build -PtargetArch=x86_64 -Pcheckstyle.ignoreFailures=false -PfailOnWarnings=true -PerrorProne=true -PskipAndroid=true -Dorg.gradle.parallel=true
/grpc-java
/grpc-java/examples/example-servlet /grpc-java

FAILURE: Build failed with an exception.

* Where:
Build file '/grpc-java/examples/example-servlet/build.gradle' line: 26

* What went wrong:
A problem occurred evaluating root project 'example-servlet'.
> Could not find method providedImplementation() for arguments [javax.servlet:javax.servlet-api:4.0.1, org.apache.tomcat:annotations-api:6.0.53] on object of type org.gradle.api.internal.artifacts.dsl.dependencies.DefaultDependencyHandler.

```

File: buildscripts/kokoro/unix.sh [hunk 1]
```diff
@@ -64,28 +64,17 @@ if [[ -z "${SKIP_TESTS:-}" ]]; then
...
-  mvn verify --batch-mode
-  popd
-  pushd examples/example-jwt-auth
-  ../gradlew build $GRADLE_FLAGS
-  mvn verify --batch-mode
-  popd
-  pushd examples/example-xds
-  ../gradlew build $GRADLE_FLAGS
-  popd
+  for f in `ls -d examples/example-*`
```
Comment (ejona86) [inline buildscripts/kokoro/unix.sh:67]: You don't need to do `ls`. You can do a glob here to list folders: `for F in examples/example-*`

File: buildscripts/kokoro/unix.sh [hunk 2]
```diff
@@ -64,28 +64,17 @@ if [[ -z "${SKIP_TESTS:-}" ]]; then
...
+  do
+     pushd "$f"
+     ../gradlew build $GRADLE_FLAGS
+     if [ -f "pom.xml" ]; then
+     # --batch-mode reduces log spam
```
Comment (ejona86) [inline buildscripts/kokoro/unix.sh:72]: Fix indentation on this and the next line.

File: examples/example-servlet/build.gradle [hunk 3]
```diff
@@ -23,7 +23,7 @@ dependencies {
             "io.grpc:grpc-servlet:${grpcVersion}",
             "io.grpc:grpc-stub:${grpcVersion}"
 
-    providedImplementation "javax.servlet:javax.servlet-api:4.0.1",
```
Comment (ejona86) [inline examples/example-servlet/build.gradle:26]: This should be `compileOnly`.

**Review by ejona86 — APPROVED** [review]

**Pull request merged**